### ISSUE TITLE
More flexible sums

### ISSUE THREAD

**Issue by yav** [GaloisInc/daedalus#117]
Different alternatives defining sums need not all define the same constructors.  

```
def Main =.
  Choose1
    Choose1
      x = UInt8
      y = UInt8
    Choose1
      x = UInt8
```

**Comment by yav** [comment]
Another example:
```
def X =
  Choose1
    x = UInt8
    y = UInt8

def Main : X =
  Choose1
    x = UInt8
```

**Comment by yav** [comment]
Should be fixed in d81166ca56634e0b39671
